cli/connhelper/commandconn/commandconn.go

```diff
@@ -90,48 +90,31 @@ func (c *commandConn) killIfStdioClosed() error {
 	return c.kill()
 }
 
-// killAndWait tries sending SIGTERM to the process before sending SIGKILL.
-func killAndWait(cmd *exec.Cmd) error {
+// kill returns nil if the command terminated, regardless to the exit status.
+func (c *commandConn) kill() error {
+	c.cmdMutex.Lock()
+	if c.cmdExited {
+		return nil
+	}
```

Comment:
It looks like we're not releasing the `.Lock()`; I think that lock is only for the `cmdExited`? If so we can release it in both cases;

```suggestion
	c.cmdMutex.Lock()
	if c.cmdExited {
		c.cmdMutex.Unlock()
		return nil
	}
	c.cmdMutex.Unlock()
```

Reply:
Yup! I meant to clean that part up a little more, thanks for pointing me in the right direction!

I need to write some more comprehensive tests for this part too, there's also https://github.com/laurazard/cli/blob/fix-connhelper-docker-example/cli/connhelper/commandconn/commandconn.go#L108 where we can hang forever if we never get another message in the channel since we try to read again and don't have any escape condition 😅